src/Backends/SystemUpdate.vala

```diff
@@ -119,11 +121,20 @@ public class SettingsDaemon.Backends.SystemUpdate : Object {
                 if (package.get_info () == SECURITY) {
                     security_updates = true;
                 }
+
+                package_ids += package.get_id ();
+            }
+
+            uint64 package_total_size = 0;
+            var results = task.get_details_sync (package_ids, null, progress_callback);
```

Comment:
just noticed this while taking a quick look, you probably shouldn't use sync here since settings daemon is also responsible for other stuff that you don't want to block

Reply:
Does this look okay? -> https://github.com/elementary/settings-daemon/pull/179/commits/b6507cf248330129c76b8d639c8dcee0b932144a